obspy/clients/nrl/client.py

```diff
@@ -0,0 +1,311 @@
+        ...     sensor_keys=['Nanometrics', 'Trillium Compact', '120 s'],
+        ...     datalogger_keys=['REF TEK', 'RT 130 & 130-SMA', '1', '200'])
+        >>> print(response)   # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+        Channel Response
+          From M/S () to COUNTS ()
+          Overall Sensitivity: 629129 defined at 0.050 Hz
+          10 stages:
+            Stage 1: PolesZerosResponseStage from M/S to V, gain: 1
+            Stage 2: ResponseStage from V to V, gain: 1
+            Stage 3: Coefficients... from V to COUNTS, gain: 629129
+            Stage 4: Coefficients... from COUNTS to COUNTS, gain: 1
+            Stage 5: Coefficients... from COUNTS to COUNTS, gain: 1
+            Stage 6: Coefficients... from COUNTS to COUNTS, gain: 1
+            Stage 7: Coefficients... from COUNTS to COUNTS, gain: 1
+            Stage 8: Coefficients... from COUNTS to COUNTS, gain: 1
+            Stage 9: Coefficients... from COUNTS to COUNTS, gain: 1
+            Stage 10: Coefficients... from COUNTS to COUNTS, gain: 1
+        """
+        resp_parser = self.get_parser(datalogger_keys, sensor_keys)
+        return resp_parser.get_response()
+
+
+class NRLDict(dict):
+    def __init__(self, nrl):
+        self._nrl = nrl
+
+    def __str__(self):
+        if len(self):
+            if self._question:
+                info = ['{} ({} items):'.format(self._question, len(self))]
+            else:
+                info = ['{} items:'.format(len(self))]
+            texts = ["'{}'".format(k) for k in sorted(self.keys())]
+            info.extend(_textwrap(", ".join(texts), initial_indent='  ',
+                                  subsequent_indent='  '))
+            return '\n'.join(_i.rstrip() for _i in info)
+        else:
+            return '0 items.'
+
+    def _repr_pretty_(self, p, cycle):  # pragma: no cover
+        p.text(str(self))
+
+    def __getitem__(self, name):
+        value = super(NRLDict, self).__getitem__(name)
+        # if encountering a not yet parsed NRL Path, expand it now
+        if isinstance(value, NRLPath):
+            value = self._nrl._parse_ini(value)
+            self[name] = value
+        return value
+
+
+class NRLPath(str):
+    pass
+
+
+class LocalNRL(NRL):
+    """
+    Subclass of NRL for accessing local copy NRL.
+    """
+    def __init__(self, root):
+        self.root = root
+        self._join = os.path.join
+        super(self.__class__, self).__init__()
+
+    def _get_cp_from_ini(self, path):
+        """
+        Returns a configparser from a path to an index.txt
+        """
+        cp = configparser.SafeConfigParser()
+        with codecs.open(path, mode='r', encoding='UTF-8') as f:
+            if sys.version_info.major == 2:  # pragma: no cover
+                cp.readfp(f)
+            else:  # pragma: no cover
+                cp.read_file(f)
+        return cp
+
+    def _read_resp(self, path):
+        # Returns Unicode string of RESP
+        with open(path, 'r') as f:
```

Comment:
is this safe? seems to be opening the file in text mode without specifying the encoding explicitly..